File: src/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

int main(int argc, char **argv) {
  ros::init(argc, argv, "talker");
  ros::NodeHandle n;
  ros::Publisher chatter_pub = 
             n.advertise<std_msgs::String>("chatter", 100);
  ros::Rate loop_rate(10);
  int count = 0;
  while (ros::ok())  {
    std_msgs::String  msg;
    std::stringstream  ss;
    ss << "Hola mundo " << count;
    msg.data = ss.str();
    std::cout << msg.data.c_str() << std::endl;
    chatter_pub.publish(msg);
    ros::spinOnce();
    loop_rate.sleep();
    ++count;
  }
  return 0;
} 

File: src/talker.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

rospy.init_node('talker', anonymous=False)
pub = rospy.Publisher('chatter', String, queue_size=100)
rate = rospy.Rate(10)
count = 0
while not rospy.is_shutdown():
    hello_str = "[Py] Hola mundo %s" % count
    count += 1
    print(hello_str)
    pub.publish(hello_str)
    rate.sleep()
